math/perfect_number.cpp

```diff
@@ -0,0 +1,58 @@
+#include <iostream>
+
+using namespace std;
+
+void Check_Perfect_Number(int number, int i, int sum) {
+
+  while (i < number) { // to check if the number itself is not less than 1 
+    if (number % i == 0) // to take out factors 
+      sum = sum + i; // adding the factors to the variable sum 
+    i++;
+  }
+
+  if (sum == number) // 
+    cout << i << " is a perfect number\n";
+  else
+    cout << i << " is not a perfect number\n";
+
+}
+
+int main() {
```

Comment:
You can automatically test your implementation with the first 7 perfect numbers: https://en.wikipedia.org/wiki/List_of_Mersenne_primes_and_perfect_numbers